### PR TITLE
Feature/access token validation

### PR DESCRIPTION
**Pull request by omkardongre**
<div id='description'>
<h3>Summary by Bito</h3>
This PR implements Auth0 JWT validation for the application. It adds configuration variables and dependencies for both backend and frontend, creates a new auth middleware for JWT validation applied to the user creation route, and updates the frontend Auth0 provider with audience configuration.
<br>
<br>
<b>Code change type</b>: Feature Addition, Security
<br>
<br>
<b>Unit tests added</b>: False
<br>
<br>
<b>Estimated effort to review (1-5, lower is better)</b>: 2
</div>

### CODE REVIEW

**Comment by omkardongre** [comment]
<div>
<h2>Code Review Agent Run #544b06</h2>
<div>
<ul>

<li>
<b>AI Based Review: </b><small>✔️</small>&nbsp;Successful
</li>

</ul>
</div>
<div>
<h3>High-level Feedback</h3>
The implementation of Auth0 authentication is a significant enhancement to the application's security. The addition of environment variable checks for AUTH0_AUDIENCE is a commendable practice. To further improve, consider adding comprehensive error handling throughout the authentication flow and including unit tests to verify the middleware's functionality. While the current changes are solid, documenting the authentication process and any potential edge cases would be beneficial for future maintenance and scalability.
</div>
<br>

<div>
<h3>Actionable Issues</h3>

<div id="issue">
<div>
:page_facing_up: backend/src/middleware/auth.ts
</div>
<details>
<summary>
Issues: Total - 1, High importance - 1
</summary>
<div>


<table width="100%">
<tbody>

<tr>
<td width="30%">
<a href ="https://github.com/omkardongre/Food-Ordering-App/pull/17/files#r1769516849">Line 6-6</a>
</td>
<td width="70%">
 <small>🔴</small> High importance - 1 &nbsp;&nbsp; 
</td>
</tr>

</tbody>
</table>

</div>
</details>
</div>

</div>




<span><i>AI Code Review powered by</i> <sub><sub><a href="https://bito.ai/" target="_blank">Bito Logo</a></sub></sub></span>

</div>

**Comment by omkardongre** [comment]
<div>
<h2>Code Review Agent Run #dbf179</h2>
<div>
<ul>

<li>
<b>AI Based Review: </b><small>✔️</small>&nbsp;Successful
</li>

<li>
<b>Security Check Analysis: </b><small>✔️</small>&nbsp;Successful.
</li>

</ul>
</div>
<div>
<h3>High-level Feedback</h3>
The implementation of Auth0 JWT validation is a significant security enhancement. The code structure is generally good, with clear separation of concerns. However, there's room for improvement in error handling. It's commendable that environment variable validation has been added, which will prevent runtime errors. To further enhance the code, consider adding more comprehensive error messages and logging for easier debugging. Overall, this is a solid addition to the application's security infrastructure.
</div>
<br>




<span><i>AI Code Review powered by</i> <sub><sub><a href="https://bito.ai/" target="_blank">Bito Logo</a></sub></sub></span>

</div>

**Comment by omkardongre** [comment]
<div>
<h2>Changelist by Bito</h2>
    <p>This pull request implements the following key changes.</p>

<div>
<table width="100%">
<tbody>
  <tr>
    <th>Key Change</th>
    <th>Files Impacted</th>
  </tr>

<tr>
<td width="30%">
    <b>Feature Improvement</b> - Auth0 JWT Validation Implementation
</td>
<td width="70%">


<div>

-&nbsp;[`.env.example`](https://github.com/omkardongre/Food-Ordering-App/pull/17/files#diff-361f2d0f55c49b270125820efb10f0d9433ced883e9d59b1f58c27b877dc2080) - Added Auth0 configuration variables
<div>

-&nbsp;[`package-lock.json`](https://github.com/omkardongre/Food-Ordering-App/pull/17/files#diff-f89763fc11c1bc48ac3c7ce8a148fbbbc5c2062393ee78664a70273d714daf5c) - Added express-oauth2-jwt-bearer dependency
<div>

-&nbsp;[`package.json`](https://github.com/omkardongre/Food-Ordering-App/pull/17/files#diff-495707834ca4b862f9acdfbac70d279023d2c059da13db59594e61ed3354fed5) - Added express-oauth2-jwt-bearer dependency
<div>

-&nbsp;[`auth.ts`](https://github.com/omkardongre/Food-Ordering-App/pull/17/files#diff-f02b89c063c2063daf35839b5e0a954d3375185b73d4214b22a59d3e93e9c5b5) - Created new auth middleware for JWT validation
<div>

-&nbsp;[`MyUserRoutes.ts`](https://github.com/omkardongre/Food-Ordering-App/pull/17/files#diff-eb6010cee74cbd710ac776c3803083e227537731d9830bce3c92983f63e8e222) - Applied JWT check to user creation route
<div>

-&nbsp;[`.env.example`](https://github.com/omkardongre/Food-Ordering-App/pull/17/files#diff-7b3b2052ac3e484bcffbb75d7ee140a9ebe7677fd7ea34ac17ca193dd3c0ecfc) - Added Auth0 audience configuration
<div>

-&nbsp;[`Auth0ProviderNavigate.tsx`](https://github.com/omkardongre/Food-Ordering-App/pull/17/files#diff-0aea0dfc8f2fae08fbda25b86c4f8d04976132edfbe49af4271dae8191ae0902) - Updated Auth0 provider with audience configuration
</div>
</td>
</tr>

</tbody>
</table>
</div>
</div>

File: backend/src/middleware/auth.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import dotenv from 'dotenv';
+import { auth } from 'express-oauth2-jwt-bearer';
+dotenv.config();
+
+export const jwtCheck = auth({
+  audience: process.env.AUTH0_AUDIENCE,
```
Comment (omkardongre) [inline backend/src/middleware/auth.ts:6]: <div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/17#issuecomment-2365075758>#544b06</a><span> - 09/21/2024, 09:16 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The code assumes that the AUTH0_AUDIENCE environment variable is set and valid without performing any checks.
</div>
<div id="fix">
<b>Fix:</b> Add a check for the existence and validity of the AUTH0_AUDIENCE environment variable before using it. Throw an error if it's not set or invalid.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,9 +1,14 @@
  import dotenv from 'dotenv';
  import { auth } from 'express-oauth2-jwt-bearer';
  dotenv.config();
 
 +if (!process.env.AUTH0_AUDIENCE) {
 +  throw new Error('AUTH0_AUDIENCE is not set in the environment variables');
 +}
 +
  export const jwtCheck = auth({
 -  audience: process.env.AUTH0_AUDIENCE,
 +  audience: process.env.AUTH0_AUDIENCE!,
    issuerBaseURL: process.env.AUTH0_IS_USER_BASE_URL,
    tokenSigningAlg: 'RS256',
  });
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

**Comment by omkardongre** [comment]
<div>
<h2>Code Review Agent Run #6405bf</h2>
<div>
<ul>

<li>
<b>AI Based Review: </b><small>✔️</small>&nbsp;Successful
</li>

<li>
<b>Security Check Analysis: </b><small>✔️</small>&nbsp;Successful.
</li>

</ul>
</div>
<div>
<h3>High-level Feedback</h3>
The implementation of Auth0 JWT validation significantly enhances the application's security. The code structure is generally good, with appropriate error handling in place. However, there are opportunities for improvement in input validation and environment variable handling. On the frontend, consider adding a trim check for the 'audience' variable to prevent potential authentication issues. For the backend, implement checks for the existence of AUTH0_AUDIENCE and AUTH0_IS_USER_BASE_URL environment variables before using them. These changes will further strengthen the security and robustness of the authentication process.
</div>
<br>




<span><i>AI Code Review powered by</i> <sub><sub><a href="https://bito.ai/" target="_blank">Bito Logo</a></sub></sub></span>

</div>

**Comment by omkardongre** [comment]
<div>
<h2>Code Review Agent Run #06ab0e</h2>
<div>
<ul>

<li>
<b>AI Based Review: </b><small>✔️</small>&nbsp;Successful
</li>

<li>
<b>Security Check Analysis: </b><small>✔️</small>&nbsp;Successful.
</li>

</ul>
</div>
<div>
<h3>High-level Feedback</h3>
The implementation of Auth0 JWT validation demonstrates a strong commitment to enhancing application security. The changes appear well-structured and comprehensive, covering both backend and frontend aspects. While no specific issues were highlighted, it would be beneficial to ensure that unit tests are added to verify the functionality of the new auth middleware. Additionally, consider documenting the new security measures for future reference and maintenance. Overall, excellent work on improving the application's security posture.
</div>
<br>




<span><i>AI Code Review powered by</i> <sub><sub><a href="https://bito.ai/" target="_blank">Bito Logo</a></sub></sub></span>

</div>
